osu.Game/Utils/GeometryUtils.cs

```diff
@@ -127,7 +138,52 @@ public static Quad GetSurroundingQuad(IEnumerable<Vector2> points)
         /// </summary>
         /// <param name="hitObjects">The hit objects to calculate a quad for.</param>
         public static Quad GetSurroundingQuad(IEnumerable<IHasPosition> hitObjects) =>
-            GetSurroundingQuad(hitObjects.SelectMany(h =>
+            GetSurroundingQuad(enumerateStartAndEndPositions(hitObjects));
+
+        /// <summary>
+        /// Returns the points that make up the convex hull of the provided points.
+        /// </summary>
+        /// <param name="points">The points to calculate a convex hull.</param>
+        public static List<Vector2> GetConvexHull(IEnumerable<Vector2> points)
```

Comment:
The convex hull of the selection is required to properly calculate the clamped scale with rotation and single axis. If you only check points on the rectangular hull of the selection, some of the points will stick out too much and over-estimate the size of the selection. The convex hull however accurately represents the width of a selection on any rotation.

Reply:
Just by chance, was the algorithm here transcribed from https://en.wikibooks.org/wiki/Algorithm_Implementation/Geometry/Convex_hull/Monotone_chain#Java? It looks _mighty_ similar.

If it _was_, then there is a bit of a licensing problem here, because the page in question is licensed under CC BY-SA, so this should be too. Notably, MIT is not listed as compatible with CC BY-SA.

Reply:
Seems like it is. I didn't realize it had such an annoying license. Now I gotta find a way to implement the exact same algorithm but more original.

Reply:
I got help from @kongehund to do a clean room implementation of the monotone chain algorithm. He has not seen any implementations of the algorithm prior, so it should be completely free of licenses.